compiler/rustc_interface/src/queries.rs

```diff
@@ -19,43 +19,68 @@ use rustc_session::{output::find_crate_name, Session};
 use rustc_span::symbol::sym;
 use rustc_span::Symbol;
 use std::any::Any;
-use std::cell::{Ref, RefCell, RefMut};
+use std::cell::{Ref, RefCell};
 use std::rc::Rc;
 use std::sync::Arc;
 
 /// Represent the result of a query.
 ///
-/// This result can be stolen with the [`take`] method and generated with the [`compute`] method.
+/// This result can be stolen once with the [`take`] method and generated with the [`compute`] method.
 ///
 /// [`take`]: Self::take
 /// [`compute`]: Self::compute
 pub struct Query<T> {
-    result: RefCell<Option<Result<T>>>,
+    result: RefCell<Option<Option<Result<T>>>>,
```

Comment:
Might be worth explaining in a comment what the difference is between `None` and `Some(None)`.